request_logging/middleware.py

```diff
@@ -87,40 +87,58 @@ def __init__(self, get_response=None):
         self.boundary = ''
 
     def __call__(self, request):
+        skip_logging_because = self._should_log_route(request)
+        if skip_logging_because:
+            return self._skip_logging(request, skip_logging_because)
+        else:
+            return self._log_cycle(request)
+
+    def _skip_logging(self, request, reason):
+        method_path = "{} {}".format(request.method, request.get_full_path())
+        no_log_context = {
+            'args': (),
+            'kwargs': {
+                'extra': {
+                    'no_logging': reason
+                },
+            },
+        }
+        self.logger.log(logging.INFO, method_path + " (not logged because %s)" % reason, no_log_context)
```

Comment:
We can slim this down to make sure no string interpolation is done unless the log level matches.

```
self.logger.info("{} {} (reason: {})", request.method, request.get_full_path(), reason)
```

Not sure of the value of `no_log_context` here unless it's for customization by end users somehow?